src/components/panningViews/panListenerView.js

```diff
@@ -0,0 +1,246 @@
+     * directions - array of directions
+     * deltas - array of deltas (same length and order as directions)
+     * Both arrays will have [x, y] - if no x or y swipe has occurred this value will be undefined
+     */
+    onSwipe: PropTypes.func,
+    /**
+     * This is were you will get notified when the pan starts
+     */
+    onPanStart: PropTypes.func,
+    /**
+     * This is were you will get notified when the pan ends
+     * The user has released all touches while this view is the responder.
+     * This typically means a gesture has succeeded
+     */
+    onPanRelease: PropTypes.func,
+    /**
+     * This is were you will get notified when the pan ends
+     * Another component has become the responder,
+     * so this gesture should be cancelled
+     */
+    onPanTerminated: PropTypes.func,
+    /**
+     * The sensitivity beyond which a pan is no longer considered a single click (default is 5)
+     */
+    panSensitivity: PropTypes.number,
+    /**
+     * The sensitivity beyond which a pan is no longer considered a drag, but a swipe (default is 1.8)
+     * Note: a pan would have to occur (i.e. the panSensitivity has already been surpassed)
+     */
+    swipeVelocitySensitivity: PropTypes.number,
+  };
+
+  static defaultProps = {
+    directions: [
+      PanningProvider.Directions.UP,
+      PanningProvider.Directions.DOWN,
+      PanningProvider.Directions.LEFT,
+      PanningProvider.Directions.RIGHT,
+    ],
+    panSensitivity: DEFAULT_PAN_SENSITIVITY,
+    swipeVelocitySensitivity: DEFAULT_SWIPE_VELOCITY,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {};
+
+    this.panResponder = PanResponder.create({
+      onStartShouldSetPanResponder: this.yes,
+      onMoveShouldSetPanResponder: this.shouldPan,
+      onStartShouldSetPanResponderCapture: this.no,
+      onMoveShouldSetPanResponderCapture: this.no,
+      onPanResponderGrant: this.handlePanStart,
+      onPanResponderMove: this.handlePanMove,
+      onPanResponderRelease: this.handlePanRelease,
+      onPanResponderTerminate: this.handlePanTerminate,
+    });
+  }
+
+  yes = () => {
+    return true;
+  };
+
+  no = () => {
+    return false;
+  };
+
+  shouldPan = (e, gestureState) => {
+    // return true if user is swiping, return false if it's a single click
+    const {dy, dx} = gestureState;
+    const {directions, panSensitivity} = this.props;
+
+    return (
+      directions &&
+      ((directions.includes(PanningProvider.Directions.UP) && dy < -panSensitivity) ||
+        (directions.includes(PanningProvider.Directions.DOWN) && dy > panSensitivity) ||
+        (directions.includes(PanningProvider.Directions.LEFT) && dx < -panSensitivity) ||
+        (directions.includes(PanningProvider.Directions.RIGHT) && dx > panSensitivity))
+    );
+  };
+
+  handlePanStart = () => {
+    const {onPanStart, context} = this.props;
+
+    if (!_.isUndefined(onPanStart)) {
+      onPanStart();
+    }
+
+    if (!_.isUndefined(context)) {
```

Comment:
Context can be undefined (when there's only a panListenerView (and no provider), but since this.props is not undefined it works fine (I guess _.invoke check for undefined on the object).
Done